include/seastar/core/scheduling_specific.hh

```diff
@@ -37,17 +37,66 @@ namespace seastar {
 namespace internal {
 
 struct scheduling_group_specific_thread_local_data {
+    using val_ptr = std::unique_ptr<void, void (*)(void*) noexcept>;
+    using cfg_ptr = std::shared_ptr<scheduling_group_key_config>;
+
+    struct specific_val {
+        val_ptr valp;
+        cfg_ptr cfg;
+
+        specific_val() : valp(nullptr, &free), cfg(nullptr) {}
+
+        specific_val(val_ptr&& valp_, const cfg_ptr& cfg_) : valp(std::move(valp_)), cfg(cfg_) {
```

Comment:
I'm probably misreading the intent, but  specific_val::cfg  and scheduling_group_specific_thread_local_data::scheduling_group_key_configs are intended to point to the _same_ shared object. But the way this constructor is defined makes it impossible to _share_ the config, the specific_val gets its _copy_ of it. Please, correct me if I'm wrong.

Reply:
Ah, yes, my bad. It's cfg_ptr